src/argilla/client/client.py

```diff
@@ -701,10 +707,12 @@ def _load_records_new_fashion(
                 vector=vector_search,
             )
             return dataset_class(results.records)
+        if fields:
+            fields.extend(["id", "inputs"])
 
         records = self.datasets.scan(
             name=name,
-            projection={"*"},
+            projection=list(set(fields)) if fields else ["*"],
```

Comment:
Parsing projection inside the `scan` method allows us to pass sets instead of lists.
```suggestion
            projection=set(fields) if fields else {"*"},
```